packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -706,6 +708,44 @@ export function createRouteManifest(
 	};
 }
 
+async function getRoutePrerenderOption(
+	route: RouteData,
+	settings: AstroSettings,
+	fsMod: typeof nodeFs | undefined,
+	logger: Logger,
+) {
+	if (route.type !== 'page' && route.type !== 'endpoint') return;
+	const localFs = fsMod ?? nodeFs;
+	const content = localFs.readFileSync(
+		fileURLToPath(new URL(route.component, settings.config.root)),
+		'utf-8',
+	);
+
+	// Check if the route is pre-rendered or not
+	const match = /^\s*export\s+const\s+prerender\s*=\s*(true|false);?/m.exec(content);
+	if (match) {
+		route.prerender = match[1] === 'true';
+	}
+
+	await runHookRouteSetup({ route, settings, logger });
+
+	if (!route.prerender) settings.buildOutput = 'server';
+}
```

Comment:
In the previous code, we added a fallback if the `route.prerender` value is unset: https://github.com/withastro/astro/blob/01b48a466e9ab58a2ad3603a2e8ed787f550b031/packages/astro/src/vite-plugin-scanner/index.ts#L97-L100

Should we also do it here too, or is it handled somewhere else?

Also, I assume `if (!route.prerender) settings.buildOutput = 'server';` is the logic that automatically switches to a server build if a non-prerendered page is detected. Do we also want to handle the inverse where if all pages are prerendered we switch to `static`? (perhaps kinda edge case, but i thought this was what the redesign enabled)

Reply:
I'm not sure (for the edge case)! I debated internally with myself, but I figure there's perhaps still a value to allow the user to force a server build? Though adapters kinda can handle this themselves too with the new option.

Reply:
If they want to force a server build (which is also an edge case itself), they can add a dummy server-only endpoint and I think that should work. Though I don't really see why someone would do that since the server build would pretty much not get used ever IIUC.